pkg/datastore/kubernetes/kubernetes.go

```diff
@@ -106,6 +106,34 @@ func (k *Datastore) GetEndpoints(clusterID string) ([]types.SubmarinerEndpoint,
 	return endpoints, nil
 }
 
+func (k *Datastore) GetClusters(colorCodes []string) ([]types.SubmarinerCluster, error) {
+	k8sClusters, err := k.client.SubmarinerV1().Clusters(k.remoteNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	clusters := []types.SubmarinerCluster{}
+
+	for _, cluster := range k8sClusters.Items {
+		for _, colorCode := range colorCodes {
+			if colorCodeMatches(colorCode, cluster.Spec.ColorCodes) {
+				clusters = append(clusters, types.SubmarinerCluster{Spec: cluster.Spec})
```

Comment:
Is there a possibility to end up with duplicate clusters in the slice here?

Reply:
I think no @Oats87. As we are using 'break' in the inner for loop once there is a match (i.e., we iterate over a cluster and when we find that it has atleast one color code, we add it to the list and stop checking further color codes). 

PS: This was a quick fix. There is some duplication which we can remove by moving them to APIs.